pkg/controller/service_instance.go

```diff
@@ -0,0 +1,183 @@
+	"github.com/atlassian/smith/pkg/util"
+	sc_v1b1 "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/bcrypt"
+	core_v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	unstructured_conversion "k8s.io/apimachinery/pkg/conversion/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	secretParametersChecksumAnnotation = smith.Domain + "/secretParametersChecksum"
+	disabled                           = "disabled"
+)
+
+// works around the fact that service catalog does not know when to send an update request if the
+// parameters section would change as the result of changing a secret referenced in parametersFrom. To
+// do this we record the contents of all referenced secrets in an annotation, compare that annotations value
+// each time a service instance is processed, and force UpdateRequests to a higher value when those secrets
+// change to trigger service catalog to send an update request.
+func (st *resourceSyncTask) forceServiceInstanceUpdates(spec *unstructured.Unstructured, actual runtime.Object, namespace string) (specRet *unstructured.Unstructured, e error) {
+	gvk := spec.GroupVersionKind()
+	gk := spec.GroupVersionKind().GroupKind()
+
+	if gk.Kind == "" || gvk.Version == "" {
+		return spec, nil
+	}
+
+	if gk.Group == sc_v1b1.GroupName && spec.GetKind() == "ServiceInstance" {
+		return st.processServiceInstance(spec, actual, namespace)
+	}
+
+	return spec, nil
+}
+
+// works by adding using an annotation to remember the previous contents of all the secrets referenced in
+// parametersFrom and forcing updateRequest to higher value when the referenced secrets change. The annotation
+// stores the hex value of the bcryted hash of the sha256 hash of the contents of all secrets referenced in
+// parametersFrom.
+//
+// The annotation is only ever added to the spec object not the actual object. The modified spec is treated as
+// if the user manually set the annotation.  This means that if the spec is updated to remove all parametersFrom
+// referenced secrets the annotation will be removed.
+//
+// If the a user actually did set the annotation manually there are two cases.
+// 1. if there are no parametersFrom referenced secrets we leave the spec untouched
+// 2. if there are referenced secrets we proceed to check and generate a new value for the annotation
+//
+// This can be disabled by adding the annotation with the value 'disabled'
+func (st *resourceSyncTask) processServiceInstance(spec *unstructured.Unstructured, actual runtime.Object, namespace string) (specRet *unstructured.Unstructured, e error) {
+	var instanceSpec sc_v1b1.ServiceInstance
+	var previousEncodedChecksum string
+	var updateCount int64
+
+	if err := unstructured_conversion.DefaultConverter.FromUnstructured(spec.Object, &instanceSpec); err != nil {
+		return nil, errors.Wrapf(err, "Failure to parse ServiceInstance spec %q", spec.GetName())
```

Comment:
By convention error messages should start from lowercase letter. Here and below

Reply:
Also, name of the object and/or resource is not needed because the error message is attached to the per-resource condition in the status. Please remove it here and below.